orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	repoFilePermPrivateRW os.FileMode = 0600
+	repoDirPermPrivateRWX os.FileMode = 0700
+)
+
+type Repo struct {
+	fileRepoDir         string
+	fileSuffix          string
+	transientFileMarker string
+
+	mu sync.Mutex
+}
+
+type repoOption func(r *Repo)
+
+func WithTransientFileMarker(marker string) repoOption {
```

Comment:
Why do we keep doing this? An exported function should accept and return exported types.

Please export `RepoOption`.

Reply:
I had initially left it unexported to leave defining functions that returned repoOptions to only be definable within this package. I guess that might have been too restrictive or unnecessary.

Reply:
May also want to consider returning an error from the function since the constructor returns an error.

Reply:
> I had initially left it unexported to leave defining functions that returned repoOptions to only be definable within this package. I guess that might have been too restrictive or unnecessary.

It's unnecessary. The reason you use options is so the implementation can access the internal details of the thing that's being constructed. The implementation of the function acting on `*Repo` must live in this package to do anything meaningful.

If you don't export the type, it's an inconsistent. It's no different than having a public constructor that returns an unexported implementation.